### PR TITLE
Update SuperRes ios app to released version ort-ext pods and some other minor refinements

### PR DESCRIPTION
**Pull request by YUNQIUGUO**
- Update SuperRes ios app to released version ort-ext pods and opset18 version model
- Refine app UI to display the output image in one screen
- Readme updates

### CODE REVIEW

**Comment by YUNQIUGUO** [comment]
I also tried to update the register custom ops using the function name approach here , but got a symbol not found error, not sure if any extra set up is required: 

https://github.com/microsoft/onnxruntime-inference-examples/blob/34de6069fd62f53f12ff2337dc12ffaae5f67172/mobile/examples/super_resolution/ios/ORTSuperResolution/ORTSuperResolution/ORTSuperResolutionPerformer.mm#L32-L34

using:
`session_options.RegisterCustomOpsUsingFunction("RegisterCustomOps");`
and latest ort-ext pod and onnxruntime-c pod

File: mobile/examples/super_resolution/ios/ORTSuperResolution/Podfile [hunk 1]
```diff
@@ -7,8 +7,8 @@ target 'ORTSuperResolution' do
   # Pods for OrtSuperResolution
   pod 'onnxruntime-c'
   
-  # Pre-release version pods
-  pod 'onnxruntime-extensions-c', '0.5.0-dev+261962.e3663fb'
+  # Release version pod for ort-ext
+  pod 'onnxruntime-extensions-c', '0.6.0'
```
Comment (edgchen1) [inline mobile/examples/super_resolution/ios/ORTSuperResolution/Podfile:11]: just use the latest?
```suggestion
  pod 'onnxruntime-extensions-c'
```

File: mobile/README.md [hunk 2]
```diff
@@ -59,4 +59,4 @@ The [Xamarin.Forms](https://dotnet.microsoft.com/apps/xamarin/xamarin-forms) exa
 The example application accomplishes the task of recovering a high resolution (HR) image from its low resolution counterpart with Ort-Extensions support for pre/post processing. Currently supports on platform Android and iOS.
 
 - [Android Super Resolution](examples/super_resolution/android)
-- [iOS Super Resolution](examples/speech_recognition/ios)
+- [iOS Super Resolution](examples/super_resolution/ios)
```
Comment (natke) [inline mobile/README.md:62]: Good spot!

File: mobile/examples/super_resolution/ios/ORTSuperResolution/ORTSuperResolution.xcodeproj/project.pbxproj [hunk 3]
```diff
@@ -7,12 +7,13 @@
 	objects = {
 
 /* Begin PBXBuildFile section */
-		510A53D6296DE390000DB268 /* pt_super_resolution_with_pre_post_processing_opset16.onnx in Resources */ = {isa = PBXBuildFile; fileRef = 510A53D5296DE390000DB268 /* pt_super_resolution_with_pre_post_processing_opset16.onnx */; };
```
Comment (natke) [inline mobile/examples/super_resolution/ios/ORTSuperResolution/ORTSuperResolution.xcodeproj/project.pbxproj:10]: Should this file be under source control?

**Review by natke — APPROVED** [review]
A couple of minor comments

File: mobile/examples/super_resolution/ios/ORTSuperResolution/ORTSuperResolution.xcodeproj/project.pbxproj (hunk 3 above)
Comment (YUNQIUGUO) [reply mobile/examples/super_resolution/ios/ORTSuperResolution/ORTSuperResolution.xcodeproj/project.pbxproj:10]: this is the older opset16 version model, do we still also want to keep that model in the project?

also fyi .pbxproj file updates are auto-generated after updating the ios app.

File: mobile/examples/super_resolution/ios/ORTSuperResolution/ORTSuperResolution.xcodeproj/project.pbxproj [hunk 4]
```diff
@@ -200,17 +201,18 @@
 			isa = PBXResourcesBuildPhase;
 			buildActionMask = 2147483647;
 			files = (
-				510A53D6296DE390000DB268 /* pt_super_resolution_with_pre_post_processing_opset16.onnx in Resources */,
 				5130D9C32968B93C009B4B88 /* Preview Assets.xcassets in Resources */,
 				5130D9C02968B93C009B4B88 /* Assets.xcassets in Resources */,
 				510A53E8296FE46F000DB268 /* cat_224x224.png in Resources */,
+				5197BE9E29A026B3004B83DE /* pytorch_superresolution_with_pre_post_processing_op18.onnx in Resources */,
```
Comment (skottmckay) [inline mobile/examples/super_resolution/ios/ORTSuperResolution/ORTSuperResolution.xcodeproj/project.pbxproj:207]: nit: 'op18' is a bit obtuse vs 'opset18'.

File: mobile/examples/super_resolution/ios/ORTSuperResolution/ORTSuperResolution/ContentView.swift [hunk 5]
```diff
@@ -38,7 +38,9 @@ struct ContentView: View {
                         Text("Output high resolution image: ").frame(width: 350, height: 40, alignment:.leading)
```
Comment (skottmckay) [inline mobile/examples/super_resolution/ios/ORTSuperResolution/ORTSuperResolution/ContentView.swift:38]: nit: would 'higher resolution image' lower user expectations of extra special magic happening?

File: mobile/examples/super_resolution/ios/ORTSuperResolution/ORTSuperResolution.xcodeproj/project.pbxproj (hunk 3 above)
Comment (skottmckay) [reply mobile/examples/super_resolution/ios/ORTSuperResolution/ORTSuperResolution.xcodeproj/project.pbxproj:10]: I'd prefer we only have the opset 18 model given that will have significantly better output.

Comment (natke) [reply mobile/examples/super_resolution/ios/ORTSuperResolution/ORTSuperResolution.xcodeproj/project.pbxproj:10]: > also fyi .pbxproj file updates are auto-generated after updating the ios app.

That's why I wondered the pbxporj file should be under version control? If it gets re-generated. Not a biggie

File: mobile/examples/super_resolution/ios/ORTSuperResolution/ORTSuperResolution.xcodeproj/project.pbxproj (hunk 4 above)
Comment (natke) [reply mobile/examples/super_resolution/ios/ORTSuperResolution/ORTSuperResolution.xcodeproj/project.pbxproj]: True! I did confuse op16 with fp16 :-)

File: mobile/examples/super_resolution/ios/ORTSuperResolution/ORTSuperResolution.xcodeproj/project.pbxproj [hunk 6]
```diff
@@ -7,12 +7,13 @@
...
 		5130D9BC2968B93A009B4B88 /* ORTSuperResolutionApp.swift in Sources */ = {isa = PBXBuildFile; fileRef = 5130D9BB2968B93A009B4B88 /* ORTSuperResolutionApp.swift */; };
 		5130D9BE2968B93A009B4B88 /* ContentView.swift in Sources */ = {isa = PBXBuildFile; fileRef = 5130D9BD2968B93A009B4B88 /* ContentView.swift */; };
 		5130D9C02968B93C009B4B88 /* Assets.xcassets in Resources */ = {isa = PBXBuildFile; fileRef = 5130D9BF2968B93C009B4B88 /* Assets.xcassets */; };
 		5130D9C32968B93C009B4B88 /* Preview Assets.xcassets in Resources */ = {isa = PBXBuildFile; fileRef = 5130D9C22968B93C009B4B88 /* Preview Assets.xcassets */; };
+		5197BEA129A0492D004B83DE /* pytorch_superresolution_with_pre_post_processing_opset18.onnx in Resources */ = {isa = PBXBuildFile; fileRef = 5197BEA029A0492D004B83DE /* pytorch_superresolution_with_pre_post_processing_opset18.onnx */; };
+		5197BEA229A0492D004B83DE /* pytorch_superresolution_with_pre_post_processing_opset18.onnx in Resources */ = {isa = PBXBuildFile; fileRef = 5197BEA029A0492D004B83DE /* pytorch_superresolution_with_pre_post_processing_opset18.onnx */; };
```
Comment (edgchen1) [inline mobile/examples/super_resolution/ios/ORTSuperResolution/ORTSuperResolution.xcodeproj/project.pbxproj:16]: why are there two entries here for pytorch_superresolution_with_pre_post_processing_opset18.onnx? was it added twice?

Comment (YUNQIUGUO) [reply mobile/examples/super_resolution/ios/ORTSuperResolution/ORTSuperResolution.xcodeproj/project.pbxproj:16]: I've only added once, not sure why there are two sections of `Resources` in the pbxproj file. but will clean it up.

/* Begin PBXResourcesBuildPhase section */
		5130D9B62968B93A009B4B88 /* Resources */ = {
			isa = PBXResourcesBuildPhase;
			buildActionMask = 2147483647;
			files = (
				510A53D6296DE390000DB268 /* pt_super_resolution_with_pre_post_processing_opset16.onnx in Resources */,
				5130D9C32968B93C009B4B88 /* Preview Assets.xcassets in Resources */,
				5130D9C02968B93C009B4B88 /* Assets.xcassets in Resources */,
				510A53E8296FE46F000DB268 /* cat_224x224.png in Resources */,
				5197BEA129A0492D004B83DE /* pytorch_superresolution_with_pre_post_processing_opset18.onnx in Resources */,
			);
			runOnlyForDeploymentPostprocessing = 0;
		};
		51AB38AA2981FA5E004E33B0 /* Resources */ = {
			isa = PBXResourcesBuildPhase;
			buildActionMask = 2147483647;
			files = (
				5197BEA229A0492D004B83DE /* pytorch_superresolution_with_pre_post_processing_opset18.onnx in Resources */,
			);
			runOnlyForDeploymentPostprocessing = 0;
		};

Comment (YUNQIUGUO) [reply mobile/examples/super_resolution/ios/ORTSuperResolution/ORTSuperResolution.xcodeproj/project.pbxproj:16]: > why are there two entries here for pytorch_superresolution_with_pre_post_processing_opset18.onnx? was it added twice?

I think one is for ORTSuperResolution and one is for ORTSuperResolutionTest. I added the model to both targets.

**Review by shahicasper — COMMENTED** [review]
apply

**Pull request merged**